Docker/podman Builds for notqmail

mbhangui opened the pull request:
# Dockerfile repository for automated builds.

This repo contains repository for Dockerfiles used for building
docker/podman images for notqmail

https://hub.docker.com/r/docker_username/notqmail

The following tags/images can be pulled by executing
the commands

## a) docker

```
docker pull docker_user/notqmail:tag
```

or

## b) podman

```
podman pull docker_user/notqmail:tag
```

Replace tag in the above command with one of the following
```
xenial     for Ubuntu 16.04
bionic     for Ubuntu 18.04
focal      for Ubuntu 20.04
centos7    for CentOS 7
centos8    for CentOS 8
debian8    for Debian 8
debian9    for Debian 9
debian10   for Debian 10
fc31       for Fedora Core 31
fc32       for Fedora Core 32
Tumbleweed for openSUSE Tumbleweed
Leap15.2   for openSUSE Leap 15.2
```
## Instructions for starting the docker/podman container
(replace podman with docker for docker operations)

### list podman images
$ podman images
```
REPOSITORY                           TAG       IMAGE ID       CREATED          SIZE
localhost/notqmail                   centos7   4bcbc7a876fe   38 minutes ago   452 MB
localhost/notqmail                   focal     d3b533841934   2 hours ago      141 MB
```

### Start the podman container

notqmail uses docker-entrypoint to execute qmail-start and start qmail-send, qmail-lspawn, qmail-rspawn.
You just need to pass any argument other than notqmail to bypass the default entrypoint.

$ podman run -d -h notqmail.org --name notqmail d3b533841934
```
08a4df5054d920cfdf8869aa777a7afc39bab19591394ea283c0c082f8b0a876
```

You can use --net host to map the container's network to the HOST
```
$ docker run --net host -d -h notqmail.org --name notqmail d3b533841934
or
$ podman run --net host -d -h notqmail.org --name notqmail d3b533841934
```

### Query the id of the container

$ podman ps
```
CONTAINER ID  IMAGE                                   COMMAND   CREATED             STATUS                 PORTS  NAMES
08a4df5054d9  docker.io/cprogrammer/notqmail:focal    notqmail  About a minute ago  Up About a minute ago         notqmail
```

### Execute an interactive shell in the container

$ podman exec -ti notqmail /bin/bash
```
root@notqmail:/#
```

### Get processlist in the container

```
root@notqmail:/# ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
qmails         1       0  0 10:10 pts/0    00:00:00 qmail-send
qmaill        35       1  0 10:10 pts/0    00:00:00 splogger qmail
root          36       1  0 10:10 pts/0    00:00:00 qmail-lspawn ./Maildir/
qmailr        37       1  0 10:10 pts/0    00:00:00 qmail-rspawn
qmailq        38       1  0 10:10 pts/0    00:00:00 qmail-clean
syslog        50       1  0 10:10 ?        00:00:00 /usr/sbin/rsyslogd
root          66       0 40 10:22 pts/1    00:00:00 bash
root          75      66  0 10:22 pts/1    00:00:00 ps -ef
root@notqmail:/#
```

### Get console logs of the container
```
$ podman logs
Warning  linking /etc/mtab to /proc/self/mounts
Your fully qualified host name is notqmail.org.
Putting notqmail.org into /var/qmail/control/me...
Putting notqmail.org into /var/qmail/control/defaulthost...
Putting notqmail.org into /var/qmail/control/envnoathost...
Putting notqmail.org into /var/qmail/control/defaultdomain...
Putting notqmail.org into /var/qmail/control/plusdomain...
Putting notqmail.org into /var/qmail/control/locals...
Putting notqmail.org into //var/qmail/control/rcpthosts...
Now qmail will refuse to accept SMTP messages except to the following...
notqmail.org
Make sure to change rcpthosts if you add hosts to locals or virtualdomains!
 * Starting enhanced syslogd rsyslogd
```

### Stop the container

$ podman stop \`podman ps -q\`
```
08a4df5054d920cfdf8869aa777a7afc39bab19591394ea283c0c082f8b0a876
```

### Clear the stopped container image

$ podman rm \`podman ps -aq\`
```
08a4df5054d920cfdf8869aa777a7afc39bab19591394ea283c0c082f8b0a876
```

## c) github respository for Dockerfile

The Dockerfile for each of the images is located in a separate subdirectory for each linux distro

**notqmail**
https://github.com/notqmail/docker/tree/master/notqmail

If you want to build the image yourself instead of using hub.docker.com,
you can use the below commands

```
$ docker build -t notqmail:focal ./Dockerfile.focal .
or
$ podman build -t notqmail:focal ./Dockerfile.focal .
```

## NOTE THIS FILE REQUIRES Editing after completing following steps ##
1. create a user on hub.docker.com (docker_user)
2. Create a repository for docker files and update point c) above
Take a look at

https://github.com/mbhangui/docker
&
https://hub.docker.com/repository/docker/cprogrammer/indimail-mta

DerDakon commented:
Looks like you started this branch on top of the opensuse build branch instead of master. You will probably need to do "git rebase -i origin/master" and then delete all lines referring to commits you don't want in this branch, save, and then "git push --force".

mbhangui commented:
> Looks like you started this branch on top of the opensuse build branch instead of master. You will probably need to do "git rebase -i origin/master" and then delete all lines referring to commits you don't want in this branch, save, and then "git push --force".

Ok. This is what I get. If I delete the 5 lines, nothing happens.  Should I change pick to drop in the 5 lines below?

```
pick df7cf93 added files for binary builds on openSUSE build service
pick beb8b8c do not install obsolete components
pick a765665 1. Change Source to https://github.com/notqmail/notqmail/releases/download/notqmail-1.08/notqmail-1.08.tar.xz in notqmail.spec 2. use _service file to fetch source directly from github 3. Build can now be triggered and fetch source directly from github 4. Revised version number to 1.08 5. Updated ChangeLogs in notqmail.spec and debian/changelog 6. Updated date/timestamp in ChangeLogs
pick 5d56e85 fix for systemd on ubuntu/debian
pick da4dfd2 fixed syntax eror and include changelog

# Rebase 43b7cb5..da4dfd2 onto 5d56e85 (5 commands)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup <commit> = like "squash", but discard this commit's log message
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified). Use -c <commit> to reword the commit message.
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
```
